pkg/util/kcp/kcp.go

```diff
@@ -20,11 +20,22 @@ import (
 	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
 )
 
-// IsKCPInRollingUpdate returns whether KCP is in rolling update.
-// IMPORTANT: This function can only be used when creating a new Machine.
+// IsKCPRollingUpdateFirstMachine returns whether KCP is creating the first rolling update CP machine.
```

Comment:
=> IsKCPRollingUpdateFirstMachine returns true if KCP is in rolling update and creating the first CP Machine.